grupos/grupo4/003-ejercicios/src/1.collection.js

```diff
@@ -0,0 +1,88 @@
+/*
+## 1. Collection
+
+Escribir una clase Collection que representa una colección de elementos. 
+Esta clase debe:
+
+- Poder construirse o vacía (sin elementos) o con una lista inicial de eltos
+- Debe tener un método `add` para poder agregar un elemento
+- Debe tener un método `delete` para poder eliminar un elemento
+- Debe tener un método `has` para poder determinar un elemento existe en la colección
+*/
+
+export class Collection {
+  constructor() {
+    this.listaElementos = [];
+  }
+
+  add(elemento) {
+    if (!this.has(elemento)) {
+      this.listaElementos.push(elemento);
+    }
+  }
+
+  has(elemento) {
+    return this.listaElementos.includes(elemento);
+  }
+
+  delete(elemento) {
+    const posicion = this.listaElementos.findIndex((x) => x == elemento);
```

Comment:
Los siguientes casos pasan como "verdaderos":

- `true == 1`
- `false == 0`
- `5 == '5'`
- `'a' == { toString() { return 'a'  } }`

Para evitar eso, hay que usar `===`:
```suggestion
    const posicion = this.listaElementos.findIndex((x) => x === elemento);
```

O usar `indexOf`:
```suggestion
    const posicion = this.listaElementos.indexOf(elemento);
```